src/server.c

```diff
@@ -447,11 +459,40 @@ static void destroy_conn(struct conn *conn)
 	sqlite3_free(conn);
 }
 
-static void stop_cb(uv_async_t *stop)
+static void handoverDoneCb(struct dqlite_node *d, int status)
+{
+	d->handover_status = status;
+	sem_post(&d->handover_done);
+}
+
+static void handoverCb(uv_async_t *handover)
+{
+	struct dqlite_node *d = handover->data;
+	int rv;
+
+	/* Nothing to do. */
+	if (d->status != DQLITE_SERVER_RUNNING) {
+		return;
+	}
+
+	/* TODO how should the server respond to client requests while in this
+	 * state? */
```

Comment:
During a `dqlite_node_handover()` request there's nothing special we need to do. If this node was not the leader, it is not serving client requests. If it was the leader, it will try to transfer leadership, and after that's done client requests will start failing and the client will figure out who the new leader is. Note that both clients and the `gateway.c` code that servers requests are prepared for losing leadership at any time anyways, regardless of `dqlite_node_handover()`, so `dqlite_node_handover()` doesn't quite change anything for them.